web/Services/CsvImportService.cs

```diff
@@ -88,84 +94,143 @@ await using (var transaction = await _dbContext.Database.BeginTransactionAsync(I
+            }
+
+            if (Boolean.TryParse(tenant.ETT, out bool boolETT) || tenant.ETT is null)
+            {
+                dbTenant.ETT = boolETT;
+            }
+            else
+            {
+                flashTypes.Add("danger");
+                flashMessages.Add(InvalidInputMessage(tenant.Name, "ETT", tenant.ETT));
+            }
+
+            if (Boolean.TryParse(tenant.UniversalCredit, out bool boolUniversalCredit) ||
+                tenant.UniversalCredit is null)
+            {
+                dbTenant.UniversalCredit = boolUniversalCredit;
+            }
+            else
+            {
+                flashTypes.Add("danger");
+                flashMessages.Add(InvalidInputMessage(tenant.Name, "UniversalCredit", tenant.UniversalCredit));
+            }
+
+            if (Boolean.TryParse(tenant.HousingBenefits, out bool boolHousingBenefits) ||
+                tenant.HousingBenefits is null)
+            {
+                dbTenant.HousingBenefits = boolHousingBenefits;
+            }
+            else
+            {
+                flashTypes.Add("danger");
+                flashMessages.Add(InvalidInputMessage(tenant.Name, "HousingBenefits", tenant.HousingBenefits));
+            }
+
+            if (Boolean.TryParse(tenant.Over35, out bool boolOver35) || tenant.Over35 is null)
+            {
+                dbTenant.Over35 = boolOver35;
+            }
+            else
+            {
+                flashTypes.Add("danger");
+                flashMessages.Add(InvalidInputMessage(tenant.Name, "Over35", tenant.Over35));
+            }
+
+            _dbContext.Tenants.Add(dbTenant);
         }
-        return (flashTypes, flashMessages);
+        
+        _dbContext.SaveChanges();
+        return (flashTypes, flashMessages, postcode);
     }
 
     private string InvalidInputMessage(string tenantName, string propName, string propValue)
     {
         _logger.LogWarning($"Invalid input for {propName} in record for {tenantName}.");
-        return $"Invalid input in record for tenant {tenantName}: '{propName}' cannot be '{propValue}' as this is the wrong data type.";
+        return $"Invalid input in record for tenant {tenantName}: '{propName}' cannot be '{propValue}' as this is the wrong data type. This input has been ignored.";
 
     }
+    
+    public (List<string> FlashTypes, List<string> FlashMessages) AddLatLonToTenantDb()
+    {
+        List<string> flashTypes = new(), flashMessages = new();
+        var tenantsWithPostcodesToConvert =
+            _dbContext.Tenants.Where(t => t.Postcode != null && t.Lat == null && t.Lon == null).ToList();
+        
+        foreach (var tenant in tenantsWithPostcodesToConvert)
+        {
+            var postcodeDbModel = _dbContext.Postcodes.FirstOrDefault(p => p.Postcode == tenant.Postcode);
+            if (postcodeDbModel == null || postcodeDbModel.Lat == null || postcodeDbModel.Lon == null)

```

Comment:
This can be merged into two :) 
```suggestion
            if (postcodeDbModel?.Lat == null || postcodeDbModel.Lon == null)

```